api/v1alpha1/drplacementcontrol_types.go

```diff
@@ -64,41 +64,89 @@ type DRPlacementControlSpec struct {
 }
 
 // DRState for keeping track of the DR placement
-// +kubebuilder:validation:Enum=Initial;Failing-over;Failed-over;Failing-back;Failed-back;Relocating;Relocated
 type DRState string
 
 // These are the valid values for DRState
 const (
-	// Initial, this is the state that will be recorded in the DRPC status
+	// Deploying, state recorded in the DRPC status to indicate that the
+	// initial deployment is in progress. Deploying means selecting the
+	// preffered cluster and creating a VRG MW for it and waiting for MW
+	// to be applied in the managed cluster
+	Deploying = DRState("Deploying")
+
+	// Deployed, this is the state that will be recorded in the DRPC status
 	// when initial deplyment has been performed successfully
-	Initial DRState = "Initial"
+	Deployed = DRState("Deployed")
 
 	// FailingOver, state recorded in the DRPC status when the failover
 	// is initiated but has not been completed yet
-	FailingOver DRState = "Failing-over"
+	FailingOver = DRState("FailingOver")
 
 	// FailedOver, state recorded in the DRPC status when the failover
 	// process has completed
-	FailedOver DRState = "Failed-over"
+	FailedOver = DRState("FailedOver")
 
 	// FailingBack, state recorded in the DRPC status when the failback
 	// is initiated but has not been completed yet
-	FailingBack DRState = "Failing-back"
+	FailingBack = DRState("FailingBack")
 
 	// FailedBack, state recorded in the DRPC status when the failback
 	// process has completed
-	FailedBack DRState = "Failed-back"
+	FailedBack = DRState("FailedBack")
+
+	// Relocating, state recorded in the DRPC status to indicate that the
+	// relocation is in progress
+	Relocating = DRState("Relocating")
+
+	// Relocated, state recorded in
+	Relocated = DRState("Relocated")
+)
 
-	// TODO: implement the relocation
-	Relocating DRState = "Relocating"
-	Relocated  DRState = "Relocated"
+const (
+	ConditionAvailable   = "Available"
+	ConditionReconciling = "Reconciling"
+)
+
+const (
+	ReasonProgressing = "Progressing"
+	ReasonCleaning    = "Cleaning"
+	ReasonSuccess     = "Success"
+	ReasonUnknown     = "Unknown"
 )
 
+// VRGResourceMeta represents the VRG resource.
+type VRGResourceMeta struct {
+	// Kind is the kind of the Kubernetes resource.
+	// +optional
+	Kind string `json:"kind"`
+
+	// Name is the name of the Kubernetes resource.
+	Name string `json:"name"`
+
+	// Name is the namespace of the Kubernetes resource.
```

Comment:
(nit) This has to be "Namespace". Not urgent and can be done later.

Reply:
It is right?